src/core_plugins/timelion/server/series_functions/es/lib/create_date_agg.js

```diff
@@ -32,7 +32,19 @@ export default function createDateAgg(config, tlConfig) {
     } else if (metric[0] && metric[1]) {
       const metricName = metric[0] + '(' + metric[1] + ')';
       dateAgg.time_buckets.aggs[metricName] = {};
-      dateAgg.time_buckets.aggs[metricName][metric[0]] = { field: metric[1] };
+      const scriptedField = scriptedFields.find(field => {
+        return field.name === metric[1];
+      });
+      if (scriptedField) {
+        dateAgg.time_buckets.aggs[metricName][metric[0]] = {
+          script: {
+            inline: scriptedField.script,
+            lang: scriptedField.lang
+          }
+        };
+      } else {
+        dateAgg.time_buckets.aggs[metricName][metric[0]] = { field: metric[1] };
+      }
```

Comment:
thoughts on extracting this boilerplate in separate module, shared with buildRequest? There's some overlap

```
      const scriptedField = scriptedFields.find(field => {
        return field.name === clause[0];
      });
      if (scriptedField) {
        termsAgg.script = {
          inline: scriptedField.script,
          lang: scriptedField.lang
        };
      } else {
        termsAgg.field = clause[0];
      }
```